### ISSUE TITLE
[Error] Docker exited with rc = 1

### ISSUE THREAD

**Issue by namho4577** [ncbi/pgap#301]
I don't know why this input.yaml doesn't work.

This is my code.

namho4577@Namho4577:~/PGAP$ ./pgap.py -r -o E_coli_result9 -g $HOME/.pgap/test_genomes/MG37/input.yaml -s 'Escherichia coli' --ignore-all-errors
The latest version of PGAP is 2024-04-27.build7426, you are using version 2023-10-03.build7061, please update.
Output will be placed in: /home/namho4577/PGAP/E_coli_result9
WARNING: memory (GiB) is less than the recommended value of 8
WARNING: memory per CPU core (GiB) is less than the recommended value of 2
PGAP failed, docker exited with rc = 1
Printing log starting from failed job:

[2024-05-04 06:51:46] DEBUG [step Cache_WGS_contig_FASTA] job input {
    "file:///pgap/pgap/vecscreen/foreign_screening.cwl#foreign_screening/Cache_WGS_contig_FASTA/input": {
        "class": "File",
        "location": "file:///pgap/user_input/input.yaml",
        "size": 98,
        "basename": "input.yaml",
        "nameroot": "input",
        "nameext": ".yaml"
    }
}
[2024-05-04 06:51:46] DEBUG [step Cache_WGS_contig_FASTA] evaluated job input to {
    "file:///pgap/pgap/vecscreen/foreign_screening.cwl#foreign_screening/Cache_WGS_contig_FASTA/input": {
        "class": "File",
        "location": "file:///pgap/user_input/input.yaml",
        "size": 98,
        "basename": "input.yaml",
        "nameroot": "input",
        "nameext": ".yaml"
    }
}
[2024-05-04 06:51:46] INFO [step Cache_WGS_contig_FASTA] start
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] initializing from file:///pgap/pgap/progs/prime_cache.cwl as part of step Cache_WGS_contig_FASTA
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] {
    "input": {
        "class": "File",
        "location": "file:///pgap/user_input/input.yaml",
        "size": 98,
        "basename": "input.yaml",
        "nameroot": "input",
        "nameext": ".yaml"
    },
    "biosource": null,
    "cache_dir": "sequence_cache",
    "ifmt": "fasta",
    "inst_mol": null,
    "molinfo": null,
    "seq_ids_name": "oseq-ids.seqids",
    "submit_block_template": null,
    "taxid": null,
    "taxon_db": null
}
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] path mappings is {
    "file:///pgap/user_input/input.yaml": [
        "/pgap/user_input/input.yaml",
        "/tmp/xozn3euv/stg2364a5fe-da2f-4e0e-8a62-4c49fd27060d/input.yaml",
        "File",
        true
    ]
}
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] command line bindings is [
    {
        "position": [
            -1000000,
            0
        ],
        "datum": "prime_cache"
    },
    {
        "prefix": "-cache",
        "position": [
            0,
            "cache_dir"
        ],
        "datum": "sequence_cache"
    },
    {
        "prefix": "-ifmt",
        "position": [
            0,
            "ifmt"
        ],
        "datum": "fasta"
    },
    {
        "prefix": "-i",
        "position": [
            0,
            "input"
        ],
        "datum": {
            "class": "File",
            "location": "file:///pgap/user_input/input.yaml",
            "size": 98,
            "basename": "input.yaml",
            "nameroot": "input",
            "nameext": ".yaml",
            "path": "/tmp/xozn3euv/stg2364a5fe-da2f-4e0e-8a62-4c49fd27060d/input.yaml",
            "dirname": "/tmp/xozn3euv/stg2364a5fe-da2f-4e0e-8a62-4c49fd27060d"
        }
    },
    {
        "prefix": "-oseq-ids",
        "position": [
            0,
            "seq_ids_name"
        ],
        "datum": "oseq-ids.seqids"
    }
]
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] initial work dir {}
[2024-05-04 06:51:46] INFO [job Cache_WGS_contig_FASTA] /tmp/rjjr7em7$ prime_cache \
    -cache \
    sequence_cache \
    -ifmt \
    fasta \
    -i \
    /tmp/xozn3euv/stg2364a5fe-da2f-4e0e-8a62-4c49fd27060d/input.yaml \
    -oseq-ids \
    oseq-ids.seqids
terminate called after throwing an instance of 'ncbi::CObjReaderParseException'
  what():  NCBI C++ Exception:
    Error: READERS(CObjReaderParseException::eNoDefline) "/export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/objtools/readers/fasta.cpp", line 396: ncbi::objects::CFastaReader::ReadOneSeq() --- CFastaReader: Input doesn't start with a defline or comment around line 1 (m_Pos = 1)
     Stack trace:
       /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/include/corelib/ncbistr.hpp:4455 CObjReaderException offset=0x0 addr=(nil)
       /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/include/objtools/readers/reader_exception.hpp:82 CParseTemplException offset=0x0 addr=(nil)
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/lib/libxobjread.so /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/include/objtools/readers/reader_exception.hpp:43 ncbi::CObjReaderParseException::CObjReaderParseException(ncbi::CDiagCompileInfo const&, ncbi::CException const*, ncbi::CObjReaderParseException::EErrCode, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, ncbi::EDiagSev) offset=0x0 addr=0x7fc5e769223a
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/lib/libxobjread.so /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/objtools/readers/fasta.cpp:393 ncbi::objects::CFastaReader::ReadOneSeq(ncbi::objects::ILineErrorListener*) offset=0x0 addr=0x7fc5e76f617b
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache :0  offset=0x0 addr=0x415d67
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache :0  offset=0x0 addr=0x41c50b
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/corelib/ncbiapp.cpp:711 ncbi::CNcbiApplicationAPI::x_TryMain(ncbi::EAppDiagStream, char const*, int*, bool*) offset=0x0 addr=0x7fc5e015c132
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/lib/libxncbi.so /export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/corelib/ncbiapp.cpp:1023 ncbi::CNcbiApplicationAPI::AppMain(int, char const* const*, char const* const*, ncbi::EAppDiagStream, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) offset=0x0 addr=0x7fc5e015f78c
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache :0  offset=0x0 addr=0x4118eb
      /usr/lib64/libc-2.17.so :0  offset=0x0 addr=0x7fc5de370554
      /panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache :0  offset=0x0 addr=0x411ad9
       :0  offset=0x0 addr=0xffffffffffffffff

Stack trace (most recent call last):
#14   Object "", at 0xffffffffffffffff, in
#13   Object "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache", at 0x411ad9, in _start
#12   Object "/usr/lib64/libc-2.17.so", at 0x7fc5de370554, in __libc_start_main
#11   Object "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache", at 0x4118eb, in main
#10   Source "/export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/corelib/ncbiapp.cpp", line 1023, in AppMain [0x7fc5e015f78c]
#9    Source "/export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/corelib/ncbiapp.cpp", line 711, in x_TryMain [0x7fc5e015c132]
#8    Object "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache", at 0x41c50b, in CPrimeCacheApplication::Run()
#7    Object "/panfs/pan1.be-md.ncbi.nlm.nih.gov/gpipe/bacterial_pipeline/system/2023-10-03.build7061/arch/x86_64/bin/prime_cache", at 0x415d67, in CPrimeCacheApplication::x_Process_Fasta(std::istream&, std::ostream&)
#6    Source "/export/home/gpipe/TeamCity/Agent3/work/427aceaa834ecbb6/ncbi_cxx/src/objtools/readers/fasta.cpp", line 393, in ReadOneSeq [0x7fc5e76f61c5]
#5    Source "../../../../gcc-7.3.0/libstdc++-v3/libsupc++/eh_throw.cc", line 93, in __cxa_throw [0x7fc5deef2222]
#4    Source "../../../../gcc-7.3.0/libstdc++-v3/libsupc++/eh_terminate.cc", line 57, in terminate [0x7fc5deef1fe0]
#3    Source "../../../../gcc-7.3.0/libstdc++-v3/libsupc++/eh_terminate.cc", line 47, in __cxa_begin_catch [0x7fc5deef1f95]
#2    Source "../../../../gcc-7.3.0/libstdc++-v3/libsupc++/vterminate.cc", line 95, in __verbose_terminate_handler [0x7fc5deef41a4]
#1    Object "/usr/lib64/libc-2.17.so", at 0x7fc5de385a77, in abort
#0    Object "/usr/lib64/libc-2.17.so", at 0x7fc5de384387, in raise
Aborted (Signal sent by tkill() 161 1000)
[2024-05-04 06:51:46] DEBUG Could not collect memory usage, job ended before monitoring began.
[2024-05-04 06:51:46] WARNING [job Cache_WGS_contig_FASTA] was terminated by signal: SIGABRT
[2024-05-04 06:51:46] WARNING [job Cache_WGS_contig_FASTA] completed permanentFail
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] outputs {
    "asn_cache": {
        "location": "file:///tmp/rjjr7em7/sequence_cache",
        "basename": "sequence_cache",
        "nameroot": "sequence_cache",
        "nameext": "",
        "class": "Directory"
    },
    "oseq_ids": {
        "location": "file:///tmp/rjjr7em7/oseq-ids.seqids",
        "basename": "oseq-ids.seqids",
        "nameroot": "oseq-ids",
        "nameext": ".seqids",
        "class": "File",
        "checksum": "sha1$9639094c36326cfaa04bff33f2b24d20616cb905",
        "size": 14,
        "http://commonwl.org/cwltool#generation": 0
    }
}
[2024-05-04 06:51:46] DEBUG [step Cache_WGS_contig_FASTA] produced output {
    "file:///pgap/pgap/vecscreen/foreign_screening.cwl#foreign_screening/Cache_WGS_contig_FASTA/asn_cache": {
        "location": "file:///tmp/rjjr7em7/sequence_cache",
        "basename": "sequence_cache",
        "nameroot": "sequence_cache",
        "nameext": "",
        "class": "Directory"
    },
    "file:///pgap/pgap/vecscreen/foreign_screening.cwl#foreign_screening/Cache_WGS_contig_FASTA/oseq_ids": {
        "location": "file:///tmp/rjjr7em7/oseq-ids.seqids",
        "basename": "oseq-ids.seqids",
        "nameroot": "oseq-ids",
        "nameext": ".seqids",
        "class": "File",
        "checksum": "sha1$9639094c36326cfaa04bff33f2b24d20616cb905",
        "size": 14,
        "http://commonwl.org/cwltool#generation": 0
    }
}
[2024-05-04 06:51:46] WARNING [step Cache_WGS_contig_FASTA] completed permanentFail
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] Removing input staging directory /tmp/xozn3euv
[2024-05-04 06:51:46] DEBUG [job Cache_WGS_contig_FASTA] Removing temporary directory /tmp/kxqk9s7i
[2024-05-04 06:51:46] INFO [workflow default_plane] completed permanentFail
[2024-05-04 06:51:46] DEBUG [workflow default_plane] outputs {
    "adaptor_blastdb_dir": null,
    "contig_ids_out": {
        "location": "file:///tmp/rjjr7em7/oseq-ids.seqids",
        "basename": "oseq-ids.seqids",
        "nameroot": "oseq-ids",
        "nameext": ".seqids",
        "class": "File",
        "checksum": "sha1$9639094c36326cfaa04bff33f2b24d20616cb905",
        "size": 14,
        "http://commonwl.org/cwltool#generation": 0
    },
    "feats": null,
    "hits": null,
    "out_cache_dir": {
        "location": "file:///tmp/rjjr7em7/sequence_cache",
        "basename": "sequence_cache",
        "nameroot": "sequence_cache",
        "nameext": "",
        "class": "Directory"
    }
}
[2024-05-04 06:51:46] DEBUG [step default_plane] produced output {
    "file:///pgap/pgap/vecscreen/vecscreen.cwl#vecscreen/default_plane/adaptor_blastdb_dir": null,
    "file:///pgap/pgap/vecscreen/vecscreen.cwl#vecscreen/default_plane/contig_ids_out": {
        "location": "file:///tmp/rjjr7em7/oseq-ids.seqids",
        "basename": "oseq-ids.seqids",
        "nameroot": "oseq-ids",
        "nameext": ".seqids",
        "class": "File",
        "checksum": "sha1$9639094c36326cfaa04bff33f2b24d20616cb905",
        "size": 14,
        "http://commonwl.org/cwltool#generation": 0
    },
    "file:///pgap/pgap/vecscreen/vecscreen.cwl#vecscreen/default_plane/feats": null,
    "file:///pgap/pgap/vecscreen/vecscreen.cwl#vecscreen/default_plane/hits": null,
    "file:///pgap/pgap/vecscreen/vecscreen.cwl#vecscreen/default_plane/out_cache_dir": {
        "location": "file:///tmp/rjjr7em7/sequence_cache",
        "basename": "sequence_cache",
        "nameroot": "sequence_cache",
        "nameext": "",
        "class": "Directory"
    }
}
[2024-05-04 06:51:46] WARNING [step default_plane] completed permanentFail
[2024-05-04 06:51:46] INFO [workflow vecscreen] completed permanentFail
[2024-05-04 06:51:46] DEBUG [workflow vecscreen] outputs {
    "adaptor_blastdb_dir": null,
    "blast_align": null,
    "calls": null,
    "contamination_feats": null,
    "filtered_align": null,
    "foreign_feats": null,
    "hits": null,
    "out_cache_dir": {
        "location": "file:///tmp/rjjr7em7/sequence_cache",
        "basename": "sequence_cache",
        "nameroot": "sequence_cache",
        "nameext": "",
        "class": "Directory"
    },
    "success": null
}
[2024-05-04 06:51:46] DEBUG [step vecscreen] produced output {
    "file:///pgap/pgap/pgap.cwl#vecscreen/success": null,
    "file:///pgap/pgap/pgap.cwl#vecscreen/calls": null
}
[2024-05-04 06:51:46] WARNING [step vecscreen] completed permanentFail
[2024-05-04 06:51:46] INFO [workflow ] completed permanentFail
[2024-05-04 06:51:46] DEBUG [workflow ] outputs {
    "calls": null,
    "cds_nucleotide_fasta": null,
    "cds_protein_fasta": null,
    "checkm_raw": null,
    "fastaval_errors": {
        "location": "file:///tmp/kpw057w_/fastaval.xml",
        "basename": "fastaval.xml",
        "nameroot": "fastaval",
        "nameext": ".xml",
        "class": "File",
        "checksum": "sha1$de6c0ec7d75456516c3b24dc8dea7e3380673a8c",
        "size": 636,
        "http://commonwl.org/cwltool#generation": 0
    },
    "final_asndisc_error_diag": null,
    "final_asnval_error_diag": null,
    "gbk": null,
    "gff": null,
    "gff_enhanced": null,
    "initial_asndisc_error_diag": null,
    "initial_asnval_error_diag": null,
    "input_fasta": {
        "class": "File",
        "location": "file:///pgap/user_input/input.yaml",
        "size": 98,
        "basename": "input.yaml",
        "nameroot": "input",
        "nameext": ".yaml"
    },
    "input_submol": {
        "class": "File",
        "location": "file:///pgap/user_input/pgap_submol_ajzoocdl.yaml",
        "size": 451,
        "basename": "pgap_submol_ajzoocdl.yaml",
        "nameroot": "pgap_submol_ajzoocdl",
        "nameext": ".yaml"
    },
    "nucleotide_fasta": null,
    "protein_fasta": null,
    "sqn": null
}
[2024-05-04 06:51:46] DEBUG Moving /tmp/kpw057w_/fastaval.xml to /pgap/output/fastaval.xml
[2024-05-04 06:51:46] DEBUG Copying /pgap/user_input/input.yaml to /pgap/output/input.yaml
[2024-05-04 06:51:46] DEBUG Copying /pgap/user_input/pgap_submol_ajzoocdl.yaml to /pgap/output/pgap_submol_ajzoocdl.yaml
[2024-05-04 06:51:46] DEBUG Removing intermediate output directory /tmp/kpw057w_
[2024-05-04 06:51:46] DEBUG Removing intermediate output directory /tmp/rjjr7em7
[2024-05-04 06:51:46] DEBUG Removing intermediate output directory /tmp/726r70ss
[2024-05-04 06:51:46] DEBUG Removing intermediate output directory /tmp/62nqoe3i
[2024-05-04 06:51:46] DEBUG Removing intermediate output directory /tmp/m6femmbq
[2024-05-04 06:51:46] WARNING Final process status is permanentFail
{
    "calls": null,
    "cds_nucleotide_fasta": null,
    "cds_protein_fasta": null,
    "checkm_raw": null,
    "fastaval_errors": {
        "location": "file:///pgap/output/fastaval.xml",
        "basename": "fastaval.xml",
        "class": "File",
        "checksum": "sha1$de6c0ec7d75456516c3b24dc8dea7e3380673a8c",
        "size": 636,
        "path": "/pgap/output/fastaval.xml"
    },
    "final_asndisc_error_diag": null,
    "final_asnval_error_diag": null,
    "gbk": null,
    "gff": null,
    "gff_enhanced": null,
    "initial_asndisc_error_diag": null,
    "initial_asnval_error_diag": null,
    "input_fasta": {
        "class": "File",
        "location": "file:///pgap/output/input.yaml",
        "size": 98,
        "basename": "input.yaml",
        "checksum": "sha1$b5c03fd011f22ad9a2d4a280ca36da6c9bfd2dae",
        "path": "/pgap/output/input.yaml"
    },
    "input_submol": {
        "class": "File",
        "location": "file:///pgap/output/pgap_submol_ajzoocdl.yaml",
        "size": 451,
        "basename": "pgap_submol_ajzoocdl.yaml",
        "checksum": "sha1$e38218f8dd835b9671a026c408adea10fba953ab",
        "path": "/pgap/output/pgap_submol_ajzoocdl.yaml"
    },
    "nucleotide_fasta": null,
    "protein_fasta": null,
    "sqn": null

**Comment by azat-badretdin** [comment]
Thank you for your report, user @namho4577 !

Your command line says: 
```
./pgap.py -r -o E_coli_result9 -g $HOME/.pgap/test_genomes/MG37/input.yaml -s 'Escherichia coli' --ignore-all-errors

```
When you are using `-g` option it requires a FASTA file, not a YAML file.

**Comment by namho4577** [comment]
Then what code should I write for the yaml file?
iOS용 Outlook<https://aka.ms/o0ukef> 다운로드
________________________________
보낸 사람: Azat Badretdin ***@***.***>
보낸 날짜: Saturday, May 4, 2024 5:57:13 PM
받는 사람: ncbi/pgap ***@***.***>
참조: charmantflower ***@***.***>; Mention ***@***.***>
제목: Re: [ncbi/pgap] [Error] Docker exited with rc = 1 (Issue #301)


Closed #301<https://github.com/ncbi/pgap/issues/301> as completed.

**Comment by azat-badretdin** [comment]
If you would like to continue using YAML file replace -s and -g parameters with YAML file. 

So this:

```
./pgap.py -r -o E_coli_result9 -g $HOME/.pgap/test_genomes/MG37/input.yaml -s 'Escherichia coli' --ignore-all-errors
```
will become this

```
./pgap.py -r --ignore-all-errors -o E_coli_result9 $HOME/.pgap/test_genomes/MG37/input.yaml 

```
